lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1856,6 +2000,22 @@ LogicalResult FIRRTLLowering::visitExpr(SpecialConstantOp op) {
                      getOrCreateIntConstant(APInt(/*bitWidth*/ 1, op.value())));
 }
 
+LogicalResult FIRRTLLowering::visitExpr(SubindexOp op) {
+  if (getLoweredValue(op) || !op.input())
+    return success();
+  if (isZeroBitFIRRTLType(op->getResult(0).getType()))
+    return setLowering(op, Value());
+
+  auto resultType = lowerType(op->getResult(0).getType());
+  Value value = getLoweredValue(op.input());
+  assert(resultType && value && "subindex type lowering failed");
+  auto iIdx = getOrCreateIntConstant(
+      log2(op.input().getType().cast<FVectorType>().getNumElements() + 1),
```

Comment:
Please don't use libm log2.  Please use llvm::Log2_64 or similar in MathExtras.h